### PR TITLE
gitter: init at 4.1.0

### PR DESCRIPTION
**Pull request by colonelpanic8**
###### Motivation for this change

Add gitter client

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ryantm** [comment]
The commit message shouldn't have a "v" before the version number.

@GrahamcOfBorg build gitter

**Comment by IvanMalison** [comment]
@ryantm done

**Review by ryantm — APPROVED** [review]

File: pkgs/applications/networking/instant-messengers/gitter/default.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    icon = pname;
+    desktopName = "Gitter";
+    genericName = meta.description;
+    categories = "Network;InstantMessaging;";
+  };
+
+  meta = with stdenv.lib; {
+    description = "Where developers come to talk";
+    downloadPage = "https://gitter.im/apps";
+    license = licenses.mit;
```
Comment (fpletz) [inline pkgs/applications/networking/instant-messengers/gitter/default.nix]: Is it really MIT-licensed? I can't find the source anywhere. Or are the binaries MIT-licensed?

Comment (ryantm) [reply pkgs/applications/networking/instant-messengers/gitter/default.nix]: Maybe https://gitlab.com/gitlab-org/gitter/webapp/blob/develop/LICENSE

Comment (IvanMalison) [reply pkgs/applications/networking/instant-messengers/gitter/default.nix]: @ryantm yeah, pretty close, although that is for the web app. This is the desktop app license: https://gitlab.com/gitlab-org/gitter/desktop/blob/master/LICENSE.md

**Comment by volth** [comment]
@GrahamcOfBorg eval

**Pull request merged**